src/GafferScene/Constraint.cpp

```diff
@@ -217,7 +230,15 @@ boost::optional<ScenePlug::ScenePath> Constraint::targetPath() const
 	ScenePath targetPath;
 	ScenePlug::stringToPath( targetPathAsString, targetPath );
 
-	if( !inPlug()->exists( targetPath ) )
+	const ScenePlug *targetScene = targetScenePlug();
+	if( !targetScene->getInput() )
+	{
+		// Backwards compatibility for time when there was
+		// no `targetScene` plug.
```

Comment:
This is just about the philosophy rather than the functionality, but defaulting to using the input scene seems like a useful default, even ignoring the previous situation?  I feel like there's precedent of other nodes where we default to the main input if a special scene isn't provided?

Reply:
I've covered this in my [reply to Tom](https://github.com/GafferHQ/gaffer/pull/4080#discussion_r555077287) above. No precedents sprung to mind though - can you name any?

Reply:
I suppose the other thing is that the general philosophy is that nodes shouldn't really care where their plugs get their values from - they're meant to be self-contained. So checking for connectedness has always felt a bit hokey to me, although there are certainly times where it's pragmatic.